src/System.Drawing.Common/src/System/Drawing/FontFamily.Unix.cs

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Drawing
+{
+    /// <summary>
+    /// Abstracts a group of type faces having a similar basic design but having certain variation in styles.
+    /// </summary>
+    public sealed partial class FontFamily : MarshalByRefObject, IDisposable
+    {
+        public override bool Equals(object obj)
+        {
+            if (obj == this)
+            {
+                return true;
+            }
+
+            // if obj = null then (obj is FontFamily) = false.
+            if (!(obj is FontFamily otherFamily))
+            {
+                return false;
+            }
+
+            // In unix FontFamily objects are not singleton so they don't share the same native pointer,
+            // the best we have to know if they are the same is FontFamily.Name which gets resolved from the native pointer.
+            return Name.Equals(otherFamily.Name, StringComparison.Ordinal);
```

Comment:
Are the names guaranteed to be non-null or could this null ref?

Reply:
I think they are guaranteed to be non-null, if the native call fails it would return an invalid status:

https://github.com/mono/libgdiplus/blob/2953bd318272b8d3ee423504a5f37c8879fe34a1/src/font.c#L539

And then when getting the name we would throw: 
https://github.com/dotnet/corefx/blob/master/src/System.Drawing.Common/src/System/Drawing/FontFamily.cs#L211